src/fluid/braginskii/bragThermalDiffusion.hpp

```diff
@@ -0,0 +1,676 @@
+                            - 0.25*(q(n,k,j - 1,i)     / q(m,k,j - 1,i)          \
+                                  + q(n,k,j - 1,i - 1) / q(m,k,j - 1,i - 1)))
+
+#define D_DZ_I_tmp(q,n,m)  (  0.25*(q(n,k + 1,j,i)     / q(m,k + 1,j,i)       \
+                                  + q(n,k + 1,j,i - 1) / q(m,k + 1,j,i - 1))  \
+                            - 0.25*(q(n,k - 1,j,i)     / q(m,k - 1,j,i)       \
+                                  + q(n,k - 1,j,i - 1) / q(m,k - 1,j,i - 1)))
+
+#define D_DX_J_tmp(q,n,m)  (  0.25*(q(n,k,j,i + 1)     / q(m,k,j,i + 1)      \
+                                  + q(n,k,j - 1,i + 1) / q(m,k,j - 1,i + 1)) \
+                            - 0.25*(q(n,k,j,i - 1)     / q(m,k,j,i - 1)      \
+                                  + q(n,k,j - 1,i - 1) / q(m,k,j - 1,i - 1)))
+
+#define D_DZ_J_tmp(q,n,m)  (  0.25*(q(n,k + 1,j,i)     / q(m,k + 1,j,i)       \
+                                  + q(n,k + 1,j - 1,i) / q(m,k + 1,j - 1,i))  \
+                            - 0.25*(q(n,k - 1,j,i)     / q(m,k - 1,j,i)       \
+                                  + q(n,k - 1,j - 1,i) / q(m,k - 1,j - 1,i)))
+
+#define D_DX_K_tmp(q,n,m)  (  0.25*(q(n,k,j,i + 1)     / q(m,k,j,i + 1)      \
+                                  + q(n,k - 1,j,i + 1) / q(m,k - 1,j,i + 1)) \
+                            - 0.25*(q(n,k,j,i - 1)     / q(m,k,j,i - 1)      \
+                                  + q(n,k - 1,j,i - 1) / q(m,k - 1,j,i - 1)))
+
+#define D_DY_K_tmp(q,n,m)  (  0.25*(q(n,k,j + 1,i)     / q(m,k,j + 1,i)       \
+                                  + q(n,k - 1,j + 1,i) / q(m,k - 1,j + 1,i))  \
+                            - 0.25*(q(n,k,j - 1,i)     / q(m,k,j - 1,i)       \
+                                  + q(n,k - 1,j - 1,i) / q(m,k - 1,j - 1,i))) \
+
+#define BX_I  Vs(BX1s,k,j,i)
+#define BY_J  Vs(BX2s,k,j,i)
+#define BZ_K  Vs(BX3s,k,j,i)
+#define BY_I (0.25*(Vs(BX2s,k,j,i) + Vs(BX2s,k,j + 1,i) \
+             + Vs(BX2s,k,j,i - 1) + Vs(BX2s,k,j + 1,i - 1)))
+#define BZ_I (0.25*(Vs(BX3s,k,j,i) + Vs(BX3s,k + 1,j,i) \
+             + Vs(BX3s,k,j,i - 1) + Vs(BX3s,k + 1,j,i - 1)))
+#define BX_J (0.25*(Vs(BX1s,k,j,i) + Vs(BX1s,k,j,i + 1) \
+             + Vs(BX1s,k,j - 1,i) + Vs(BX1s,k,j - 1,i + 1)))
+#define BZ_J (0.25*(Vs(BX3s,k,j,i) + Vs(BX3s,k + 1,j,i) \
+             + Vs(BX3s,k,j - 1,i) + Vs(BX3s,k + 1,j - 1,i)))
+#define BX_K (0.25*(Vs(BX1s,k,j,i) + Vs(BX1s,k,j,i + 1) \
+             + Vs(BX1s,k - 1,j,i) + Vs(BX1s,k - 1,j,i + 1)))
+#define BY_K (0.25*(Vs(BX2s,k,j,i) + Vs(BX2s,k,j + 1,i) \
+             + Vs(BX2s,k - 1,j,i) + Vs(BX2s,k - 1,j + 1,i)))
+
+
+// This function computes the thermal Flux and stores it in hydro->fluxRiemann
+// (this avoids an extra array)
+template <Limiter limTemplate>
+void BragThermalDiffusion::AddBragDiffusiveFluxLim(int dir, const real t,
+                                                const IdefixArray4D<real> &Flux) {
+#if HAVE_ENERGY == 1
```

Comment:
This results in an additional HAVE_ENERGY cjeck in the calcParabolicFlux routine as the function AddBragDiffusiveFlux is no longer defined when HAVE_ENERGY is false.

Reply:
Two things:

- I'm not in favour of adding a switch on HAVE_ENERGY in CalcParabolicFlux (because I'm trying to get rid of these #ifdef everywhere, so it goes against this effort)
- We do need to have `AddBragDiffuiveFlux` to be defined because the code might call this, no matter whether HAVE_ENERGY is set or not

The reason for this #ifdef was that PRS and ENG macros were *not* defined when HAVE_ENERGY was not set, and hence the thermal diffusion routines would not compile. This was in v1.

The code now relies mostly on templates, PRS and ENG are always defined, so I *think* that this `#ifdef` can be removed altogether (also true for isotropic thermal diffusion, actually)

Reply:
Ok then I'll remove it in the braginskii case, and in the isotropic case as well